### ISSUE TITLE
Behaviour of `mode='drop'` is inconsistent between `jnp.array.at.add` and `jax.ops.segment_sum`

### ISSUE THREAD

**Issue by danielegrattarola** [google/jax#17387]
### Description

The behaviour when setting `mode='drop'` is inconsistent between `jnp.array.at.add` and `jax.ops.segment_sum`. 

In one case, -1 is treated as in-bounds, in the other it isn't. 

```python
i = jnp.array([0, 1, 2, 3, 4, -1, -1, -1, -1, -1])

a = jnp.zeros(10)
print(a.at[i].add(1, mode='drop'))  
# >>> [1. 1. 1. 1. 1. 0. 0. 0. 0. 5.]

a = jnp.ones(10)
print(jax.ops.segment_sum(a, i, num_segments=10, mode='drop'))
# >>> [1. 1. 1. 1. 1. 0. 0. 0. 0. 0.]
```

There doesn't seem to be a standard expected behaviour for how to treat -1, judging by the docs [here](https://jax.readthedocs.io/en/latest/jax.lax.html#jax.lax.GatherScatterMode), and the behaviour of `segment_sum` is correctly documented [here](https://jax.readthedocs.io/en/latest/_autosummary/jax.ops.segment_sum.html), but I think the two functions should have the same behaviour.

### What jax/jaxlib version are you using?

v0.4.15

**Comment by jakevdp** [comment]
Thanks - I believe it is intended that negative indices are not supported in `jax.ops`, but we should probably make that more clear in the documentation.

**Comment by jakevdp** [comment]
From the [`jax.segment_sum` docs](https://jax.readthedocs.io/en/latest/_autosummary/jax.ops.segment_sum.html):

> By default, values outside of the range `[0, num_segments)` are dropped and do not contribute to the sum.

**Comment by danielegrattarola** [comment]
Yes the behavior of segment_sum is documented correctly, but the docs of both segment_sum and `at` refer to jax.lax.GatherScatterMode, implying that the behavior is the same.

(sorry for no formatting, I am writing on a phone)

**Comment by jakevdp** [comment]
Oh, right they both use `GatherScatterMode`, but `jnp.ndarray.at` supports negative indices, while `jax.ops.segment_sum` does not. I think this is the intended behavior for both.
